eole/modules/multi_headed_attn.py

```diff
@@ -649,35 +652,32 @@ def forward(
                         ],
                         dim=-2,
                     )
-                    if (
-                        self.position_encoding_type == PositionEncodingType.Rotary
-                        and start_pos + 32 >= self.rope.size(0)
-                    ):
-                        # Resize rotary embeddings.
-                        # We take a margin of 32 tokens as the kv_cache
-                        #   is incremented by 32 tokens every 32 tokens.
-                        self.rope, self.cos, self.sin = rotaryembeddings(
-                            self.rotary_dim,
-                            maxseqlen=(start_pos + 2048),
-                            base=self.rotary_theta,
-                            device=self.rope.device,
-                        )
-
                 if sliding_window > 0 and key.size(2) > sliding_window:
                     self.layer_cache[1]["keys"] = self.layer_cache[1]["keys"][
                         :, :, 1:, :
                     ]
                     self.layer_cache[1]["values"] = self.layer_cache[1]["values"][
                         :, :, 1:, :
                     ]
+                if position_embeddings is not None:
+                    cos = (
```

Comment:
not sure why we compute cos and sin here, then are self.cos and self.sin still needed ?

Reply:
It's not really computed, just formatted. Previously we had 3 objects (rope, cos, sin) moving around, but cos and sin are only used here, and easily deduced from rope.
self.cos/self.sin are not there anymore, unless I missed something.